pkg/apis/v1/nodeclaim_conversion.go

```diff
@@ -101,16 +115,19 @@ func (in *NodeClaimSpec) convertFrom(ctx context.Context, v1beta1nc *v1beta1.Nod
 	defaultNodeClassGVK := injection.GetNodeClasses(ctx)[0]
 	nodeclassGroupVersion, err := schema.ParseGroupVersion(v1beta1nc.NodeClassRef.APIVersion)
 	if err != nil {
-		return err
+		return "", err
 	}
 	in.NodeClassRef = &NodeClassReference{
 		Name:  v1beta1nc.NodeClassRef.Name,
 		Kind:  lo.Ternary(v1beta1nc.NodeClassRef.Kind == "", defaultNodeClassGVK.Kind, v1beta1nc.NodeClassRef.Kind),
 		Group: lo.Ternary(v1beta1nc.NodeClassRef.APIVersion == "", defaultNodeClassGVK.Group, nodeclassGroupVersion.Group),
 	}
 
-	// Need to implement Kubelet Conversion
-	return nil
+	kubelet, err := json.Marshal(v1beta1nc.Kubelet)
+	if err != nil {
+		return "", err
+	}
+	return string(kubelet), nil
```

Comment:
Is there any key validation we need to do to make sure it can be inputted as an annotation?

Reply:
CEL validation will catch that, and deny the update

Reply:
I'm wondering if the set of characters that we can marshal to are mutually exclusive from the disallowed characters in a kubernetes annotation. If they're not mutually exclusive, then we could potentially run into a case where a kubelet config never can be converted to an annotation.

Reply:
The values of annotation do not contain validation. This means that the characters that we can marshal to are mutually exclusive and should not cause a problem. https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/

Reply:
Ah gotcha. Just tried everything myself. Thanks!